File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson04Polymorphism/Lab/pr03_shapes/Rectangle.java
package Lesson04Polymorphism.Lab.pr03_shapes;

public class Rectangle extends Shape {

    private double height;
    private double width;

    public Rectangle(double height, double width) {
        this.setHeight(height);
        this.setWidth(width);
        this.calculatePerimeter();
        this.calculateArea();
    }

    public double getHeight() {
        return height;
    }

    private void setHeight(double height) {
        if (height < 0d) {
            throw new IllegalArgumentException();
        }
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    private void setWidth(double width) {
        if (height < 0d) {
            throw new IllegalArgumentException();
        }
        this.width = width;
    }

    @Override
    protected void calculatePerimeter() {
        super.setPerimeter((this.height + this.width) * 2d);
    }

    @Override
    protected void calculateArea() {
        super.setArea(this.height * this.width);
    }
}

File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/DrawingTool/Rectangle.cs
using System;

public class Rectangle : Shape
{
    public virtual int Length { get; set; }
    public virtual int Height { get; set; }

    public Rectangle(int length, int height)
    {
        this.Height = height;
        this.Length = length;
    }

    public override void Draw()
    {
        Console.WriteLine($"|{new string('-', this.Length)}|");

        for (int i = 0; i < this.Height - 2; i++)
        {
            Console.WriteLine($"|{new string (' ', this.Length)}|");
        }

        Console.WriteLine($"|{new string('-', this.Length)}|");
    }
}